### ISSUE TITLE
Add symlink to generalize macOS library pathname

### ISSUE THREAD

**Issue by rprimmer** [python/cpython#118052]
# Feature or enhancement

### Proposal:

For v3.12.3, the library pathname on macOS is `/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages`. When running pyinstaller it's necessary to designate the pathname to packages when creating an executable from a python script. Since this pathname is specific to 3.12, it's not obvious how to create a generic path to python packages.

There is a symlink in `/Library/Frameworks/Python.framework/Versions` for `Current -> 3.12`, but there isn't a symlink in `/Library/Frameworks/Python.framework/Versions/Current/lib`.  I manually created a symlink in this dir `Current -> python3.12`, but this will be removed with the next update. 

So the request for enhancement is to have this symlink (i.e., `Current -> release number` in dir `/Library/Frameworks/Python.framework/Versions/Current/lib`) be part of the standard install. 

### Has this already been discussed elsewhere?

This is a minor feature, which does not need previous discussion elsewhere

**Comment by ronaldoussoren** [comment]
I'm -1 on this, the links in Python.framework are there to conform to platform conventions a "lib" or "bin" symlink doesn't conform to these conventions. Having the symlink can also lead to confusing behaviour when having multiple Python versions installed at the same time (as I commonly do). 

Furthermore the library path can easily be retrieved using `sysconfig.get_paths()`, and that works on all platforms. 

BTW. Are you sure you need to specify a path to the site-packages directory? I'd expect that pyinstaller can find that on its own.

**Comment by rprimmer** [comment]
Hi Ronald, thank you for your response, it cleared up some things.

Let me just start by making sure I wasn't ambiguous in my first note. I only created a symlink in `/Library/Frameworks/Python.framework/Versions/Current/lib`; the symlink `Current -> 3.12` in dir `/Library/Frameworks/Python.framework/Versions`
was already present as part of the python package install that I downloaded (for macOS) from python.org.

However, your question as to why I would even have need for adding the path in the first place made me realize I must have something wrong in my environment if others aren't seeing this issue on macOS.

After some digging, it turns out to be a toolchain issue. On this system I use python from the package downloaded from python.org. But there was some leftover stuff from homebrew on the same system. Homebrew downloads python3 as a dependency for some of its other packages, whether you explicitly download it yourself or not.

Once I uninstalled the brew pyinstaller it worked fine without having to set `pathex=['/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages']` in the project.spec file.

Thank you for your clear explanation. It pointed me in the right direction to figure out the toolchain conflict.
